settings: set theme on back-end, update across clients

jamesacklin opened the pull request:
Fixes TLON-4118 by storing the theme on the back-end and syncing across clients.

- Adds proper settings bucket support for themes
- Syncs across devices by listening for changes to the theme setting
- Appropriately handles the "auto" (null) theme selection
- Adds logging for theme setting + errors

https://github.com/user-attachments/assets/47412726-be97-4db6-9e9c-5bcffc28d389

jamesacklin commented on packages/app/features/settings/ThemeScreen.tsx:
```diff
@@ -48,29 +50,93 @@ export function ThemeScreen(props: Props) {
+    const syncAndUpdateTheme = async () => {
+      try {
+        // First sync with backend to get latest settings
+        await syncSettings();
```
Yes - addressed

jamesacklin commented:
@dnbrwstr Thanks for the feedback. I've:

- Moved all theme updating logic into the shared store via updateTheme
- Removed the redundant syncing in ThemeScreen.tsx that's already handled by subscription
- Centralized theme conversion logic with helper functions
- Got rid of all the stray comments and logs

dnbrwstr reviewed (approved):
Looking good, few more nits!

jamesacklin commented:
@dnbrwstr OK! I...

- Got rid of the duplicate helper declaration
- Moved the theme mutation out of state
- Set up a new DB hook for settings *in general*, figure this might be useful down the line
- Changed the app provider to consume that hook, then fall back to direct access

Appreciate the help here!

jamesacklin commented on packages/app/provider/index.tsx:
```diff
@@ -1,87 +1,129 @@
+  tamaguiProps,
+}: {
+  children: React.ReactNode;
+  tamaguiProps: Omit<TamaguiProviderProps, 'config'>;
```
Was following the pattern above where we omit 'config' from TamaguiProviderProps because we're providing our own config import and passing it in directly.

jamesacklin commented on packages/shared/src/urbit/settings.ts:
```diff
@@ -1,6 +1,8 @@
 import { DisplayMode, SortMode } from './channel';
 
-export type Theme = 'light' | 'dark' | 'auto';
+export type AppTheme = ThemeName | 'auto';
```
Not stoked on the duplicate declaration but gets us around the circular dependency issue

jamesacklin commented on packages/shared/src/store/settingsActions.ts:
```diff
@@ -6,6 +8,8 @@ import * as logic from '../logic';
 import { withRetry } from '../logic';
 import { TalkSidebarFilter } from '../urbit';
 
+export type AppTheme = ThemeName | 'auto';
```
Same here

dnbrwstr reviewed (approved):
Nice! Lgtm.
